pymeasure/instruments/keysight/keysight81160A.py

```diff
@@ -0,0 +1,208 @@
+
+
+class Keysight81160AChannel(Agilent33500Channel):
+    shape_values = WF_SHAPES
+    shape_get_command = ":FUNC{ch}?"
+    shape_set_command = ":FUNC{ch} %s"
+
+    frequency_values = FREQUENCY_RANGE
+    frequency_get_command = ":FREQ{ch}?"
+    frequency_set_command = ":FREQ{ch} %f"
+
+    amplitude_values = AMPLITUDE_RANGE
+    amplitude_get_command = ":VOLT{ch}?"
+    amplitude_set_command = ":VOLT{ch} %f"
+
+    amplitude_unit_get_command = ":VOLT{ch}:UNIT?"
+    amplitude_unit_set_command = ":VOLT{ch}:UNIT %s"
+
+    offset_values = OFFSET_RANGE
+    offset_get_command = ":VOLT{ch}:OFFS?"
+    offset_set_command = ":VOLT{ch}:OFFS %f"
+
+    voltage_high_values = VOLTAGE_HIGH_RANGE
+    voltage_high_get_command = ":VOLT{ch}:HIGH?"
+    voltage_high_set_command = ":VOLT{ch}:HIGH %f"
+
+    voltage_low_values = VOLTAGE_LOW_RANGE
+    voltage_low_get_command = ":VOLT{ch}:LOW?"
+    voltage_low_set_command = ":VOLT{ch}:LOW %f"
+
+    square_dutycycle_values = SQUARE_DUTYCYCLE_RANGE
+    square_dutycycle_set_command = ":FUNC{ch}:SQU:DCYC %f"
+    square_dutycycle_get_command = ":FUNC{ch}:SQU:DCYC?"
+
+    ramp_symmetry_set_command = ":FUNC{ch}:RAMP:SYMM %f"
+    ramp_symmetry_get_command = ":FUNC{ch}:RAMP:SYMM?"
+
+    pulse_period_values = PULSE_PERIOD_RANGE
+    pulse_period_set_command = ":PULS:PER{ch} %e"
+    pulse_period_get_command = ":PULS:PER{ch}?"
+
+    pulse_hold_set_command = ":PULS:HOLD{ch} %s"
+    pulse_hold_get_command = ":PULS:HOLD{ch}?"
+
+    pulse_width_values = PULSE_WIDTH_RANGE
+    pulse_width_set_command = ":PULS:WIDT{ch} %e"
+    pulse_width_get_command = ":PULS:WIDT{ch}?"
+
+    pulse_dutycycle_values = PULSE_DUTYCYCLE_RANGE
+    pulse_dutycycle_set_command = ":PULS:DCYC{ch} %f"
+    pulse_dutycycle_get_command = ":PULS:DCYC{ch}?"
+
+    pulse_transition_values = PULSE_TRANSITION_RANGE
+    pulse_transition_set_command = ":FUNC{ch}:PULS:TRAN %e"
+    pulse_transition_get_command = ":FUNC{ch}:PULS:TRAN?"
+
+    output_load_values = IMPEDANCE_RANGE
+    output_load_set_command = ":OUTP{ch}:LOAD %e"
+    output_load_get_command = ":OUTP{ch}:LOAD?"
+
+    burst_state_values = STATES
+    burst_state_set_command = ":BURS{ch}:STAT %s"
+    burst_state_get_command = ":BURS{ch}:STAT?"
+
+    burst_mode_set_command = ":BURS{ch}:MODE %s"
+    burst_mode_get_command = ":BURS{ch}:MODE?"
+
+    burst_period_values = BURST_PERIOD
+    burst_period_set_command = ":BURS{ch}:INT:PER %e"
+    burst_period_get_command = ":BURS{ch}:INT:PER?"
+
+    burst_ncycles_values = BURST_NCYCLES
+    burst_ncycles_set_command = ":BURS{ch}:NCYC %d"
+    burst_ncycles_get_command = ":BURS{ch}:NCYC?"
+
+    limit_state = Instrument.control(
+        ":VOLT{ch}:LIM:STAT?",
+        ":VOLT{ch}:LIM:STAT %s",
+        """ Control the limit state (string).""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, "on": 1, "ON": 1, False: 0, "off": 0, "OFF": 0},
+        dynamic=True,
+    )
```

Comment:
For such a boolean property, a name like `limit_state_enabled` is better, then you may just use boolean values (and not a combination of strings and bools).